src/Runner.Worker/JobExtension.cs

```diff
@@ -242,6 +243,59 @@ public void FinalizeJob(IExecutionContext jobContext, Pipelines.AgentJobRequestM
                     context.Start();
                     context.Debug("Starting: Complete job");
 
+                    // Evaluate job outputs
+                    if (message.JobOutputs != null)
+                    {
+                        try
+                        {
+                            context.Output($"Evaluate and set job outputs");
+
+                            // Populate env context for each step
+                            Trace.Info("Initialize Env context for evaluating job outputs");
+#if OS_WINDOWS
+                            var envContext = new DictionaryContextData();
+#else
+                            var envContext = new CaseSensitiveDictionaryContextData();
+#endif
+                            context.ExpressionValues["env"] = envContext;
+                            foreach (var pair in context.EnvironmentVariables)
+                            {
+                                envContext[pair.Key] = new StringContextData(pair.Value ?? string.Empty);
+                            }
+
+                            Trace.Info("Initialize steps context for evaluating job outputs");
+                            context.ExpressionValues["steps"] = context.StepsContext.GetScope(context.ScopeName);
+
+                            var templateTrace = context.ToTemplateTraceWriter();
```

Comment:
it looks like you dont use this local var

Reply:
forgot to remove this after rebase on master :)